Fix duplicated channel's name error

agarciamontoro opened the pull request:
#### Summary
Return the correct error when the channel's name is already taken. Also remove a useless condition check.

(I know, I know, this is not frequencies related, but I needed a break!)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53756

#### Screenshots
--

#### Release Note
```release-note
Fixed the error returned by `PUT /api/v4/channels/{channelid}` when the provided name already exists in the team.
```

agarciamontoro commented:
/update-branch

agnivade commented:
> (I know, I know, this is not frequencies related, but I needed a break!)

LOL ... Now back to work!

agnivade commented on server/channels/store/sqlstore/channel_store.go:
```diff
@@ -780,10 +780,7 @@ func (s SqlChannelStore) updateChannelT(transaction *sqlxTxWrapper, channel *mod
 		if IsUniqueConstraintError(err, []string{"Name", "channels_name_teamid_key"}) {
 			dupChannel := model.Channel{}
 			s.GetReplicaX().Get(&dupChannel, "SELECT * FROM Channels WHERE TeamId = :TeamId AND Name= :Name AND DeleteAt > 0", map[string]any{"TeamId": channel.TeamId, "Name": channel.Name})
```
We can remove this line now. `dupChannel` is not used anymore.

agnivade commented on server/channels/api4/channel_test.go:
```diff
@@ -144,6 +144,37 @@ func TestUpdateChannel(t *testing.T) {
 	client := th.Client
 	team := th.BasicTeam
 
+	t.Run("update channel with another channel's name", func(t *testing.T) {
```
We should actually write a store layer test for this. That way it's easier to assert the actual error type and the field. The client layer test asserts the error string which is brittle and can change.

isacikgoz commented on server/channels/store/sqlstore/channel_store.go:
```diff
@@ -780,10 +780,7 @@ func (s SqlChannelStore) updateChannelT(transaction *sqlxTxWrapper, channel *mod
 		if IsUniqueConstraintError(err, []string{"Name", "channels_name_teamid_key"}) {
 			dupChannel := model.Channel{}
 			s.GetReplicaX().Get(&dupChannel, "SELECT * FROM Channels WHERE TeamId = :TeamId AND Name= :Name AND DeleteAt > 0", map[string]any{"TeamId": channel.TeamId, "Name": channel.Name})
-			if dupChannel.DeleteAt > 0 {
-				return nil, store.NewErrInvalidInput("Channel", "Id", channel.Id)
-			}
-			return nil, store.NewErrInvalidInput("Channel", "Id", channel.Id)
```
This looks interesting :)

isacikgoz reviewed (approved):
LGTM, 1/5 testing on store layer.

agarciamontoro commented:
Sorry for the force-push, I had forgotten I ran `update-branch` here and things got messy in my local branch. Anyway, re-requesting review after addressing the comments. The test is now much cleaner, thanks :ok_hand:

agnivade reviewed (approved):
Nicely done

streamer45 commented:
@agarciamontoro Could the same issue affect the insert path? (see `saveChannelT`).
